rest_api/controller/file_upload.py

```diff
@@ -1,98 +1,61 @@
-    SPLIT_RESPECT_SENTENCE_BOUNDARY
-from haystack.document_store.elasticsearch import ElasticsearchDocumentStore
-from haystack.file_converter.pdf import PDFToTextConverter
-from haystack.file_converter.txt import TextConverter
-from haystack.preprocessor.preprocessor import PreProcessor
+from fastapi import APIRouter, UploadFile, File, Form, HTTPException
 
+from haystack.pipeline import Pipeline
+from rest_api.config import PIPELINE_YAML_PATH, FILE_UPLOAD_PATH, INDEXING_PIPELINE_NAME
 
 logger = logging.getLogger(__name__)
 router = APIRouter()
 
+try:
+    INDEXING_PIPELINE = Pipeline.load_from_yaml(Path(PIPELINE_YAML_PATH), pipeline_name=INDEXING_PIPELINE_NAME)
+except KeyError:
+    INDEXING_PIPELINE = None
+    logger.info("Indexing Pipeline not found in the YAML configuration. File Upload API will not be available.")
 
-document_store = ElasticsearchDocumentStore(
-    host=DB_HOST,
-    port=DB_PORT,
-    username=DB_USER,
-    password=DB_PW,
-    index=DB_INDEX,
-    label_index=DB_INDEX_FEEDBACK,
-    scheme=ES_CONN_SCHEME,
-    ca_certs=None,
-    verify_certs=False,
-    text_field=TEXT_FIELD_NAME,
-    search_fields=SEARCH_FIELD_NAME,
-    embedding_dim=EMBEDDING_DIM,
-    embedding_field=EMBEDDING_FIELD_NAME,
-    excluded_meta_data=EXCLUDE_META_DATA_FIELDS,  # type: ignore
-    faq_question_field=FAQ_QUESTION_FIELD_NAME,
-    create_index=CREATE_INDEX,
-    update_existing_documents=UPDATE_EXISTING_DOCUMENTS,
-    similarity=VECTOR_SIMILARITY_METRIC
-)
 
 os.makedirs(FILE_UPLOAD_PATH, exist_ok=True)  # create directory for uploading files
 
 
 @router.post("/file-upload")
-def upload_file_to_document_store(
+def file_upload(
     file: UploadFile = File(...),
-    remove_numeric_tables: Optional[bool] = Form(REMOVE_NUMERIC_TABLES),
-    remove_whitespace: Optional[bool] = Form(REMOVE_WHITESPACE),
-    remove_empty_lines: Optional[bool] = Form(REMOVE_EMPTY_LINES),
-    remove_header_footer: Optional[bool] = Form(REMOVE_HEADER_FOOTER),
-    valid_languages: Optional[List[str]] = Form(VALID_LANGUAGES),
-    split_by: Optional[str] = Form(SPLIT_BY),
-    split_length: Optional[int] = Form(SPLIT_LENGTH),
-    split_overlap: Optional[int] = Form(SPLIT_OVERLAP),
-    split_respect_sentence_boundary: Optional[bool] = Form(SPLIT_RESPECT_SENTENCE_BOUNDARY),
+    meta: Optional[str] = Form("null"),  # JSON serialized string
+    remove_numeric_tables: Optional[bool] = Form(None),
+    remove_whitespace: Optional[bool] = Form(None),
+    remove_empty_lines: Optional[bool] = Form(None),
+    remove_header_footer: Optional[bool] = Form(None),
+    valid_languages: Optional[List[str]] = Form(None),
+    split_by: Optional[str] = Form(None),
+    split_length: Optional[int] = Form(None),
+    split_overlap: Optional[int] = Form(None),
+    split_respect_sentence_boundary: Optional[bool] = Form(None),
 ):
+    if not INDEXING_PIPELINE:
+        raise HTTPException(status_code=501, detail="Indexing Pipeline is not configured.")
```

Comment:
`HTTPStatus.NOT_IMPLEMENTED`

Reply:
HTTP 501 represents "not implemented". Do you mean changing 501 to NOT_IMPLEMENTED for code readability?